### PR TITLE
Fix #739 - disambiguate Packages and Dependencies

### LINKED ISSUE #739: Confused meaning of "Packages" field in .languages.java options
### What happened?

We accidentally overloaded two meanings on the `packages` field in the language specific configuration for the schemas, see: 

```
// PackageInfo represents a Java language-specific info for a package.
type PackageInfo struct {
	Packages    map[string]string `json:"packages,omitempty"`
	BasePackage string            `json:"basePackage"`

	// If set to "gradle" generates a basic set of Gradle build files.
	BuildFiles string `json:"buildFiles"`
}
```

One use case is https://github.com/pulumi/pulumi-kubernetes/pull/2055/files#diff-8dfd74ae4dfdfd04f5c226c8e65a2d3060d6761b45252ba7a757c22492d99311R206 controlling friendly names for the generated versioned packages.

```
            "packages": {
                "admissionregistration.k8s.io/v1": "admissionregistration.v1",
            }
```

Another use case is specifying dependencies on other packages:

https://github.com/pulumi/pulumi-java/blob/main/pkg/codegen/java/templates_gradle.go#L75

You might want to say here:

```
            "packages": {
                "com.pulumi:azuread": "5.22.0",
            }
```

And have `pulumi-java-gen` generate a build file that includes a dependency on that package.

We should disambiguate this and add doc strings to explain what the options do.

I think the right thing to do is keep packages for the original use case, and fixup the version use case to use a different field.

### Steps to reproduce

-

### Expected Behavior

-

### Actual Behavior

-

### Versions used

_No response_

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by t0yv0**
# Description

Fixes #739, #749

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Review by stack72 — APPROVED** [review]
This is way better :) Thank you!!

File: pkg/codegen/java/importer.go [hunk 1]
```diff
@@ -21,56 +21,11 @@ import (
 	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
 )
 
-const defaultBasePackage = "com.pulumi."
```
Comment (pawelprazak) [inline pkg/codegen/java/importer.go:24]: nice!

File: pkg/codegen/java/package_info.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+// These options affect how GeneratePackage generates and lays out
+// Java SDKs for Pulumi Packages.
+//
+// The options are typically set in the .language.java section of
+// the Schema, but can also be overridden during code generation:
+//
+//     pulumi-java-gen generate --override overrides.json
+//
+// See https://www.pulumi.com/docs/guides/pulumi-packages/schema/#language-specific-extensions
+type PackageInfo struct {
```
Comment (pawelprazak) [inline pkg/codegen/java/package_info.go:31]: thank you for documenting this so well :)

File: pkg/codegen/java/gen_test.go [hunk 3]
```diff
@@ -124,9 +124,8 @@ func TestGeneratePackage(t *testing.T) {
 func generatePackage(tool string, pkg *schema.Package, extraFiles map[string][]byte) (map[string][]byte, error) {
 	pkgInfo := PackageInfo{
 		BuildFiles: "gradle",
-		Packages: map[string]string{
+		Dependencies: map[string]string{
```
Comment (pawelprazak) [inline pkg/codegen/java/gen_test.go:127]: I like this new name

**Comment by pawelprazak** [comment]
Thank you for fixing this so fast! :)

**Pull request merged**